components/api-resource-mgt/org.wso2.carbon.identity.api.resource.collection.mgt/src/main/java/org/wso2/carbon/identity/api/resource/collection/mgt/APIResourceCollectionManagerImpl.java

```diff
@@ -0,0 +1,199 @@
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static org.wso2.carbon.identity.api.resource.collection.mgt.util.APIResourceCollectionManagementUtil.handleServerException;
+
+/**
+ * API Resource Collection Manager Implementation.
+ */
+public class APIResourceCollectionManagerImpl implements APIResourceCollectionManager {
+
+    private static final APIResourceCollectionManagerImpl INSTANCE;
+    private static final APIResourceCollectionMgtConfigBuilder configBuilder;
+    private static Map<String, APIResourceCollection> apiResourceCollectionMap;
+
+    static {
+        configBuilder = APIResourceCollectionMgtConfigBuilder.getInstance();
+        apiResourceCollectionMap = configBuilder.getApiResourceCollectionMgtConfigurations();
+        INSTANCE = new APIResourceCollectionManagerImpl();
+    }
+
+    private APIResourceCollectionManagerImpl() {
+    }
+
+    public static APIResourceCollectionManagerImpl getInstance() {
+
+        return INSTANCE;
+    }
+
+    /**
+     * Get API Resource Collections.
+     *
+     * @param filter Filter expression.
+     * @return API resource collection search result.
+     * @throws APIResourceCollectionMgtException If an error occurred while retrieving API Resource Collections.
+     */
+    @Override
+    public APIResourceCollectionSearchResult getAPIResourceCollections(String filter, List<String> requiredAttributes,
+                                                                       String tenantDomain)
+            throws APIResourceCollectionMgtException {
+
+        Map<String, APIResourceCollection> filteredMap =
+                FilterUtil.filterAPIResourceCollections(apiResourceCollectionMap, filter);
+        if (requiredAttributes != null &&
+                requiredAttributes.contains(APIResourceCollectionManagementConstants.API_RESOURCES)) {
+            for (Map.Entry<String, APIResourceCollection> entry : filteredMap.entrySet()) {
+                entry.setValue(populateAPIResourcesForCollection(entry.getValue(), tenantDomain));
+            }
+        }
+        return new APIResourceCollectionSearchResult(new ArrayList<>(filteredMap.values()));
+    }
+
+    /**
+     * Get API Resource Collection by id.
+     *
+     * @param collectionId Collection id.
+     * @return API resource collection.
+     * @throws APIResourceCollectionMgtException If an error occurred while retrieving API Resource Collection.
+     */
+    @Override
+    public APIResourceCollection getAPIResourceCollectionById(String collectionId, String tenantDomain)
+            throws APIResourceCollectionMgtException {
+
+        APIResourceCollection apiResourceCollection = apiResourceCollectionMap.get(collectionId);
+        if (apiResourceCollection == null) {
+            return null;
+        }
```

Comment:
do we need this check?

Reply:
Moved this inside the populateAPIResourcesForCollection method